lsp/roslyn_ls.lua

```diff
@@ -0,0 +1,161 @@
+--- borrowed some source from: https://github.com/seblj/roslyn.nvim
+-- config that activates keymaps and enables snippet support
+---comment
+
+local uv = vim.uv
+local fs = vim.fs
+
+---@param client vim.lsp.Client
+---@param target string
+local function on_init_sln(client, target)
+  vim.notify('Initializing: ' .. target, vim.log.levels.INFO, { title = 'roslyn_ls' })
+  ---@diagnostic disable-next-line: param-type-mismatch
+  client:request('solution/open', {
```

Comment:
You can use `client:notify` and then not pass an empty function as last argument here and for the `project/open`

```lua
client:notify("solution/open", {
    solution = vim.uri_from_fname(solution),
})
```